src/components/rules/rulesetAddEdit.tsx

```diff
@@ -195,6 +202,27 @@ const rulesetSchema = z
           fatal: true,
         });
     }
+
+    const names = rules.map((rule) => rule.name);
+    const isDupe = names.find((item, idx) => names.indexOf(item) !== idx);
+    if (isDupe) {
+      console.log("dupe", isDupe);
+      rules.forEach((r, i) => {
+        if (r.name === isDupe) {
+          ctx.addIssue({
+            code: z.ZodIssueCode.custom,
+            message: "Rules cannot have the same name",
+            path: [`rules[${i}][name]`],
+            fatal: true,
+          });
+        }
+      });
+      return false;
+    }
+
+    if (!isDupe) {
+      return true;
+    }
```

Comment:
Couldn't refine the schema before extending, so added refinement here since the extended schemas cover all paths (`data_sources === rabbitmq and kafka`)